### PR TITLE
added lazy loading to carousel on product page

### PR DESCRIPTION
**Pull request by qdz90**

### CODE REVIEW

File: sections/product-template.liquid [hunk 1]
```diff
@@ -176,7 +176,22 @@
...
+                    height="{{ section.settings.modal_image_height }}"
+                    >
+                {% else %}
+                  <img 
+                    data-src="{{ image.src | img_url: modal_image_size, crop: 'center' }}" 
+                    class="lazyload d-block w-100 h-auto" alt="{{ image.alt | escape }}" 
+                    width="{{ section.settings.modal_image_width }}" 
+                    height="{{ section.settings.modal_image_height }}"
+                    >
+                {% endif %}
```
Comment (qdz90) [inline sections/product-template.liquid:194]: - first image contains a src
- rest of the images contains data-src

File: sections/product-template.liquid [hunk 2]
```diff
@@ -322,6 +337,11 @@
         $('#product-photo-gallery-carousel').on('slide.bs.carousel', function (event) {
           // adjust counter on carousel slide
           counterEl.text(`${Number(event.to) + 1}/${count}`)
+
+          // lazy load slide image
+          var image = $(event.relatedTarget).find('img[data-src]');
+          image.attr('src', image.data('src'));
+          image.removeAttr('data-src');
```
Comment (qdz90) [inline sections/product-template.liquid:344]: - on slide change: update from data-src to src

File: sections/product-template.liquid [hunk 3]
```diff
@@ -176,7 +176,22 @@
...
               <div class="carousel-item {% if forloop.index0 == 0 %}active{% endif %}">
-                <img src="{{ image.src | img_url: modal_image_size, crop: 'center' }}" class="d-block w-100 h-auto" alt="{{ image.alt | escape }}" width="{{ section.settings.modal_image_width }}" height="{{ section.settings.modal_image_height }}">
+                
+                {% if forloop.index0 == 0 %}
+                  <img 
+                    src="{{ image.src | img_url: modal_image_size, crop: 'center' }}" 
+                    class="lazyload d-block w-100 h-auto" alt="{{ image.alt | escape }}" 
+                    width="{{ section.settings.modal_image_width }}" 
+                    height="{{ section.settings.modal_image_height }}"
+                    >
```
Comment (alvinkonda) [inline sections/product-template.liquid:186]:
```suggestion
                  >
```

File: sections/product-template.liquid [hunk 4]
```diff
@@ -176,7 +176,22 @@
...
+                {% else %}
+                  <img 
+                    data-src="{{ image.src | img_url: modal_image_size, crop: 'center' }}" 
+                    class="lazyload d-block w-100 h-auto" alt="{{ image.alt | escape }}" 
+                    width="{{ section.settings.modal_image_width }}" 
+                    height="{{ section.settings.modal_image_height }}"
+                    >
```
Comment (alvinkonda) [inline sections/product-template.liquid:193]:
```suggestion
                  >
```

File: sections/product-template.liquid (hunk 2 above)
Comment (alvinkonda) [inline sections/product-template.liquid:344]:
```suggestion
          // Lazyload images inside carousel
          var image = $(event.relatedTarget).find('img[data-src]')
          image.attr('src', image.data('src'))
          image.removeAttr('data-src')
```

File: sections/product-template.liquid [hunk 5]
```diff
@@ -176,7 +176,22 @@
             
             {% for image in product.images %}
               <div class="carousel-item {% if forloop.index0 == 0 %}active{% endif %}">
-                <img src="{{ image.src | img_url: modal_image_size, crop: 'center' }}" class="d-block w-100 h-auto" alt="{{ image.alt | escape }}" width="{{ section.settings.modal_image_width }}" height="{{ section.settings.modal_image_height }}">
+                
+                {% if forloop.index0 == 0 %}
+                  <img 
+                    src="{{ image.src | img_url: modal_image_size, crop: 'center' }}" 
+                    class="lazyload d-block w-100 h-auto" alt="{{ image.alt | escape }}"
```
Comment (alvinkonda) [inline sections/product-template.liquid:183]:
```suggestion
                    class="d-block w-100 h-auto" alt="{{ image.alt | escape }}" 
```

**Pull request merged**